### ISSUE TITLE
CA2000 false positive when returning collection expressions

### ISSUE THREAD

**Issue by kasthack** [dotnet/roslyn-analyzers#7508]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2013): `Call System.IDisposable.Dispose on object created by '...' before all references to it are out of scope (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000)`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [9.0.101](https://dotnet.microsoft.com/en-us/download/dotnet/9.0)

### Describe the bug

Returning a collection expression of `IDisposable` triggers CA2000 even though it shouldn't.

### Steps To Reproduce

csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net9.0</TargetFramework>
    <AnalysisLevel>latest</AnalysisLevel>
    <AnalysisMode>all</AnalysisMode>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>
</Project>

```

code:

```csharp
using System.Data;
namespace CA2000;
public static class Repro
{
    public static DataColumn[] FalsePositive() =>
    [
        new DataColumn(),                       // <- triggers CA2000
    ];
    public static DataColumn[] NotTriggered() => new []
    {
        new DataColumn()                        // works fine
    };
}
```

### Expected behavior

`FalsePositive` method shouldn't trigger CA2000.

### Actual behavior

```
1>C:\source\playground\src\Playground\CA2000\Program.cs(8,9,8,25): warning CA2000: Call System.IDisposable.Dispose on object created by 'new DataColumn()' before all references to it are out of scope (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000)
```

## Additional context

\-
